cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/cluster/AbstractClusterOperations.java

```diff
@@ -126,10 +138,61 @@ public ClusterDiffResult diff() {
         });
     }
 
-    public abstract void create(String namespace, String name, Handler<AsyncResult<Void>> handler);
+    protected abstract void create(String namespace, String name, Handler<AsyncResult<Void>> handler);
+
+    public void createByName(String namespace, ConfigMap add)   {
+        String name = add.getMetadata().getName();
+        log.info("Adding {} cluster {}", clusterDescription, name);
+
+        create(namespace, name, res -> {
+            if (res.succeeded()) {
+                log.info("{} cluster added {}", clusterDescription, name);
+            }
+            else {
+                log.error("Failed to add {} cluster {}.", clusterDescription, name);
+            }
+        });
+    }
+
+    protected abstract void delete(String namespace, String name, Handler<AsyncResult<Void>> handler);
+
+    public void deleteByLabel(String namespace, R dep)   {
```

Comment:
Is the `dep` name supposed to mean "deployment" ? Thinking about that I thought that it was `Deployment` in K8S/OS terms ? Maybe a `res`, `resource` are better ?

Reply:
You're right, but @scholzj's comment have made me rethink this anyway, so this code will be disappearing when I push another change.